### ISSUE TITLE
Unable to delete custom options in the Admin panel if the list of options takes more than one page

### ISSUE THREAD

**Issue by PiimegaOy** [magento/magento2#8213]
### Preconditions
Magento 2.1.0 (happens also in 2.1.2)
PHP 7.0.8 & MySQL 5.6.31-0ubuntu0.14.04.2

### Steps to reproduce

1. Create a product in Magento Admin Panel
2. Add 20+ custom options and save product. (It should produce more than one page for options)
3. Try to delete an option on the first page

### Expected result

1. custom option should be deleted

### Actual result

1. New empty option is added to the end of the list and saving is not possible

Here is explanation of the bug from one our developers:

file: magento/module-ui/view/base/web/js/dynamic-rows/dynamic-rows.js method "deleteRecord"
This method is called when we are trying to delete an option in the admin panel. At the end of the first condition if(this.deletePropery), there is the following code

```
childs = this.getChildItems();

if (childs.length > this.elems().length) {
    this.addChild(false, childs[childs.length - 1][this.identificationProperty], false);
}
```
Let's assume that we have deleted option 15.

The getChildItems() method will returns 19 options from the first page, i.e. 1 - 20, but without 15.
However, the this.elems() method return 20 options, 1 - 21 excluding 15, i.e. it appends one more option from the next page.

For this reason, the addChild() method is called, and new custom option prevents saving the product.

**Comment by NotAndD** [comment]
> The getChildItems() method will returns 19 options from the first page, i.e. 1 - 20, but without 15.
However, the this.elems() method return 20 options, 1 - 21 excluding 15, i.e. it appends one more option from the next page.

Let me understand, if `childs = this.getChildItems();` value is 19 and `this.elems().length` value is 20, the if condition you posted is `19 > 20`, that's false and the addChild() is not called. Or did I get it wrong?

**Comment by PiimegaOy** [comment]
@ziliquas 

Sorry, there was a mistake in the explanation

It's the other way around. This.elems().length is 19 and this.getChildItems().length is 20 and thats the reason addChild() method is called

**Comment by drestine** [comment]
Hi @veloraven, any update on this issue?

**Comment by AlexWorking** [comment]
Thank's for contacting us on this issue. We created an appropriate internal ticket MAGETWO-69973. After the bug is fixed You'll be immediately notified.

**Comment by magento-engcom-team** [comment]
@PiimegaOy, thank you for your report.
The issue is already fixed in develop branch, 2.2.0

**Comment by developerRitesh** [comment]
> @PiimegaOy, thank you for your report.
> The issue is already fixed in develop branch, 2.2.0

Hi Can anyone provide link for the fix or reference ?? i need to fix it on my setup without upgrading the magento to 2.2
